### PR TITLE
CIRC-1488 Truncate loan due date during checkout only if there is a recall request on the same item

### PR DESCRIPTION
**Pull request by Mykyta-Varenyk**
Resolves: https://issues.folio.org/browse/CIRC-1488. 
## Purpose
Currently, if the TLR feature is enabled, we recall the loan even if there is a recall request for another item of the same instance. This leads to loan flag wasDueDateChangedByRecall being true, and as a result due date and loan action are not being updated when creating a recall ILR. 
## Approach
This PR fixes the problem by looking for a request for a specific item in the queue.

### CODE REVIEW

File: src/test/java/api/requests/RequestsAPILoanHistoryTests.java [hunk 1]
```diff
@@ -43,6 +43,41 @@ void creatingRecallRequestChangesTheOpenLoanForTheSameItem() {
       is("recallrequested"));
   }
 
+  @Test
+  void checkOutShouldNotRecallLoanIfRecallRequestExistsForAnotherItemOfTheSameInstanceIfTlrIsEnabled() {
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/requests/RequestsAPILoanHistoryTests.java:47]: Recall or truncate?

File: src/test/java/api/requests/RequestsAPILoanHistoryTests.java [hunk 2]
```diff
@@ -43,6 +43,41 @@ void creatingRecallRequestChangesTheOpenLoanForTheSameItem() {
...
+    var steve = usersFixture.steve();
+    var charlotte = usersFixture.charlotte();
+
+    ItemResource firstItem = items.get(0);
+    checkOutFixture.checkOutByBarcode(firstItem, steve);
+    requestsClient.create(new RequestBuilder()
+      .recall()
+      .forItem(firstItem)
+      .withPickupServicePointId(pickupServicePointId)
+      .by(charlotte));
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/requests/RequestsAPILoanHistoryTests.java:60]: Can we use `RequestsFixture#recallItem` instead?

File: src/test/java/api/requests/RequestsAPILoanHistoryTests.java [hunk 3]
```diff
@@ -43,6 +43,41 @@ void creatingRecallRequestChangesTheOpenLoanForTheSameItem() {
...
+    var secondItem = items.get(1);
+    var loanAfterCheckOut = checkOutFixture.checkOutByBarcode(secondItem, charlotte).getJson();
+    assertThat("Action was not updated", loanAfterCheckOut.getString("action"),
+      is("checkedout"));
+
+    requestsClient.create(new RequestBuilder()
+      .recall()
+      .forItem(secondItem)
+      .withPickupServicePointId(pickupServicePointId)
+      .by(steve)).getJson().getJsonObject("loan").getString("id");
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/requests/RequestsAPILoanHistoryTests.java:71]: why do you get the id here?

File: src/test/java/api/requests/RequestsAPILoanHistoryTests.java [hunk 4]
```diff
@@ -43,6 +43,41 @@ void creatingRecallRequestChangesTheOpenLoanForTheSameItem() {
...
+
+    var loanAfterRecall = loansFixture.getLoanById(
+      UUID.fromString(loanAfterCheckOut.getString("id"))).getJson();
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/requests/RequestsAPILoanHistoryTests.java:74]:
```suggestion
      loanAfterCheckOut.getId()).getJson();
```

File: src/test/java/api/requests/RequestsAPILoanHistoryTests.java (hunk 2 above)
Comment (Mykyta-Varenyk) [reply src/test/java/api/requests/RequestsAPILoanHistoryTests.java]: Refactored

File: src/test/java/api/requests/RequestsAPILoanHistoryTests.java (hunk 4 above)
Comment (Mykyta-Varenyk) [reply src/test/java/api/requests/RequestsAPILoanHistoryTests.java]: Refactored

File: src/test/java/api/requests/RequestsAPILoanHistoryTests.java (hunk 1 above)
Comment (Mykyta-Varenyk) [reply src/test/java/api/requests/RequestsAPILoanHistoryTests.java]: Renamed

**Review by OleksandrVidinieiev — APPROVED** [review]

**Pull request merged**